### PR TITLE
Add net_interface declarative module

### PR DESCRIPTION
**Pull request by ganeshrn**
##### SUMMARY

*  Add net_interface module
*  Add junos_interface implementation module
*  Other minor changes

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

net_interface
junos_interface

##### ANSIBLE VERSION

```
2.4
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/network/junos/junos_interface.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    decription:
+      - Maximum size of transmit packet
+  duplex:
+    description:
+      - Interface link status
+    default: full
+    choices: ['full', 'half', 'auto']
+  tx_rate:
+    description:
+      - Transmit rate
```
Comment (gundalow) [inline lib/ansible/modules/network/junos/junos_interface.py]: `Transmit rate in bytes.`?

File: lib/ansible/modules/network/junos/junos_interface.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  rx_rate:
+    descriptiom:
+      - Recevier rate
```
Comment (gundalow) [inline lib/ansible/modules/network/junos/junos_interface.py]: `Receive rate in bytes.`?

File: lib/ansible/modules/network/junos/junos_interface.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+options:
+  name:
+    description:
+      - Name of the Interface.
+  description:
+    description:
+      - Description of Interface.
+  enabled:
+    description:
+      - Operational status of the interface link
```
Comment (gundalow) [inline lib/ansible/modules/network/junos/junos_interface.py]: Missing `.`

File: lib/ansible/modules/network/junos/junos_interface.py [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'core'}
```
Comment (gundalow) [inline lib/ansible/modules/network/junos/junos_interface.py:24]: We need to check this

**Review by privateip — APPROVED** [review]

File: lib/ansible/modules/network/junos/junos_interface.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+  purge:
+    description:
+      - Purge Interfaces not defined in the collections parameter.
+        This applies only for logical interface
+    default: no
+  state:
+    description:
+      - State of the Interface.
+    default: present
+    choices: ['present', 'absent', active, suspend]
```
Comment (gundalow) [inline lib/ansible/modules/network/junos/junos_interface.py]: Missing quotes

File: lib/ansible/modules/network/junos/junos_interface.py (hunk 3 above)
Comment (ganeshrn) [reply lib/ansible/modules/network/junos/junos_interface.py]: Fixed

File: lib/ansible/modules/network/junos/junos_interface.py (hunk 5 above)
Comment (ganeshrn) [reply lib/ansible/modules/network/junos/junos_interface.py]: Fixed

File: lib/ansible/modules/network/junos/junos_interface.py (hunk 4 above)
Comment (ganeshrn) [reply lib/ansible/modules/network/junos/junos_interface.py:24]: @gundalow It is currently same for other declarative modules already added.

**Pull request merged**